api/src/test/java/com/epam/pipeline/controller/datastorage/DataStorageControllerTest.java

```diff
@@ -0,0 +1,1374 @@
+        assertResponse(mvcResult, nfsDataStorage, DatastorageCreatorUtils.NFS_STORAGE_TYPE);
+    }
+
+    @Test
+    public void shouldFailFindDataStorageByPathForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(FIND_BY_PATH_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldFindS3BucketStorageByPath() throws Exception {
+        Mockito.doReturn(s3Bucket).when(mockStorageApiService).loadByPathOrId(TEST);
+
+        final MvcResult mvcResult = performRequest(get(FIND_BY_PATH_URL).param(ID_PARAM, TEST));
+
+        Mockito.verify(mockStorageApiService).loadByPathOrId(TEST);
+        assertResponse(mvcResult, s3Bucket, DatastorageCreatorUtils.S3_BUCKET_TYPE);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldFindAzureBlobStorageByPath() throws Exception {
+        Mockito.doReturn(azureBlob).when(mockStorageApiService).loadByPathOrId(TEST);
+
+        final MvcResult mvcResult = performRequest(get(FIND_BY_PATH_URL).param(ID_PARAM, TEST));
+
+        Mockito.verify(mockStorageApiService).loadByPathOrId(TEST);
+        assertResponse(mvcResult, azureBlob, DatastorageCreatorUtils.AZURE_BLOB_TYPE);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldFindGSBucketStorageByPath() throws Exception {
+        Mockito.doReturn(gsBucket).when(mockStorageApiService).loadByPathOrId(TEST);
+
+        final MvcResult mvcResult = performRequest(get(FIND_BY_PATH_URL).param(ID_PARAM, TEST));
+
+        Mockito.verify(mockStorageApiService).loadByPathOrId(TEST);
+        assertResponse(mvcResult, gsBucket, DatastorageCreatorUtils.GS_BUCKET_TYPE);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldFindNFSDataStorageByPath() throws Exception {
+        Mockito.doReturn(nfsDataStorage).when(mockStorageApiService).loadByPathOrId(TEST);
+
+        final MvcResult mvcResult = performRequest(get(FIND_BY_PATH_URL).param(ID_PARAM, TEST));
+
+        Mockito.verify(mockStorageApiService).loadByPathOrId(TEST);
+        assertResponse(mvcResult, nfsDataStorage, DatastorageCreatorUtils.NFS_STORAGE_TYPE);
+    }
+
+    @Test
+    public void shouldFailGetDataStorageItemsForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(String.format(DATASTORAGE_ITEMS_URL, ID)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldGetDataStorageFolder() throws Exception {
+        final DataStorageListing dataStorageListing = DatastorageCreatorUtils.getDataStorageListing();
+        final List<DataStorageFolder> folders = Collections.singletonList(folder);
+        dataStorageListing.setResults(Collections.singletonList(folder));
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add(PATH, TEST);
+        params.add(SHOW_VERSION, FALSE_AS_STRING);
+        Mockito.doReturn(dataStorageListing).when(
+                mockStorageApiService).getDataStorageItems(ID, TEST, false, null, null
+        );
```

Comment:
Could you please use the following syntax? It was a discussion on leaving a closing parenthesis on a new line way before and it was considered as not a java-ish syntax :)

```java
Mockito.doReturn(dataStorageListing)
    .when(mockStorageApiService).getDataStorageItems(ID, TEST, false, null, null);
```